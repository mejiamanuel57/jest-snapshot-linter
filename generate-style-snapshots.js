const React = require("react");
const { StyleSheet, Text } = require("react-native");
const TestRenderer = require("react-test-renderer");
const {
  addSerializers,
  compose,
  hoistStyleTransform,
  justChildren,
  replaceTransform,
  stylePrinter
} = require("@times-components/jest-serializer");
const styled = require("styled-components").default;
const Enzyme = require("enzyme");
const mount = require("enzyme").mount;
const Adapter = require("enzyme-adapter-react-16");

Enzyme.configure({ adapter: new Adapter() });

addSerializers(
  expect,
  compose(
    stylePrinter,
    replaceTransform({
      div: justChildren,
      Foo: justChildren
    }),
    hoistStyleTransform
  )
);

require("jest-styled-components");

it("1. snapshot for style block", () => {
  const styles = StyleSheet.create({
    test: {
      color: "blue"
    }
  });

  const testInstance = TestRenderer.create(
    <Text foo="bar" style={styles.test}>
      Some text
    </Text>
  );

  expect(testInstance).toMatchSnapshot();
});

it("2. snapshot for style block with array", () => {
  const StyledH1 = styled.h1`
    font-size: 1.5em;
    text-align: center;
    color: palevioletred;
  `;

  const Foo = () => <div>{[<StyledH1 key="1" />, <StyledH1 key="2" />]}</div>;

  const wrapper = mount(<Foo />);

  expect(wrapper).toMatchSnapshot();
});

it("3. snapshot for style block and styled components", () => {
  const styles = StyleSheet.create({
    test: {
      color: "blue"
    }
  });

  const StyledH1 = styled.h1`
    font-size: 1.5em;
    text-align: center;
    color: palevioletred;
  `;

  const wrapper = mount(<StyledH1 style={styles.test} />);

  expect(wrapper).toMatchSnapshot();
});
