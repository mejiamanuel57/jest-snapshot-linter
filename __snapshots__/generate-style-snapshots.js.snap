// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1. snapshot for style block 1`] = `
<style>
{
  "IS1": {
    "color": "blue",
  },
}
</style>

<Text
  accessible={true}
  allowFontScaling={true}
  className="IS1"
  ellipsizeMode="tail"
  foo="bar"
>
  Some text
</Text>
`;

exports[`2. snapshot for style block with array 1`] = `
.c0 {
  font-size: 1.5em;
  text-align: center;
  color: palevioletred;
}

Array [
  <styled.h1>
    <h1
      className="c0"
    />
  </styled.h1>,
  <styled.h1>
    <h1
      className="c0"
    />
  </styled.h1>,
]
`;

exports[`3. snapshot for style block and styled components 1`] = `
.c0 {
  font-size: 1.5em;
  text-align: center;
  color: palevioletred;
}

<style>
{
  "IS1": {
    "color": "blue",
  },
  "IS2": {
    "color": "blue",
  },
}
</style>

<styled.h1
  className="IS2"
>
  <h1
    className="c0 IS1"
  />
</styled.h1>
`;
